tests/FileUpload/FacebookResumableUploaderTest.php

```diff
@@ -0,0 +1,222 @@
+{
+    public function send($url, $method, $body, array $headers, $timeOut)
+    {
+        if (preg_split("/=/", explode('&', $body)[0])[1] == 'start') {
+            return $this->mockStart();
+        }
+
+        if (preg_split("/=/", explode('&', $body)[0])[1] == 'finish') {
+            return $this->mockFinish();
+        }
+
+        return $this->mockTransfer();
+    }
+
+    public function mockStart()
+    {
+        $header = [
+            "Content-Type" => "application/json; charset=UTF-8",
+            "Date" => "Mon, 10 Aug 2015 07:14:04 GMT",
+            "facebook-api-version" => "v2.3",
+            "WWW-Authenticate" => 'OAuth "Facebook Platform" "invalid_token" "Error validating access token',
+        ];
+
+        $body = '{"error":{"message":"Error validating access token: Session has expired on Monday, ' .
+            '10-Aug-15 01:00:00 PDT. The current time is Monday, 10-Aug-15 01:14:23 PDT.",' .
+            '"type":"OAuthException","code":190,"error_subcode":463}}';
+
+        return new GraphRawResponse($header, $body, 400);
+    }
+
+    public function mockTransfer()
+    {
+        $header = [
+            "Content-Type" => "application/json; charset=UTF-8",
+            "Date" => "Mon, 10 Aug 2015 07:14:04 GMT",
+            "facebook-api-version" => "v2.3",
+            "WWW-Authenticate" => 'OAuth "Facebook Platform" "invalid_token" "Error validating access token',
+        ];
+
+        $body = '{"error":{"message":"Error validating access token: Session has expired on Monday, ' .
+            '10-Aug-15 01:00:00 PDT. The current time is Monday, 10-Aug-15 01:14:23 PDT.",' .
+            '"type":"OAuthException","code":190,"error_subcode":1363033}}';
+
+        return new GraphRawResponse($header, $body, 400);
+    }
+
+    public function mockFinish()
+    {
+        $header = [
+            "Content-Type" => "application/json; charset=UTF-8",
+            "Date" => "Mon, 10 Aug 2015 07:14:04 GMT",
+            "facebook-api-version" => "v2.3",
+        ];
+
+        $body = '{"success":true}';
+
+        return new GraphRawResponse($header, $body, 200);
+    }
+}
+
+
+class FacebookResumableUploaderTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var FacebookApp
+     */
+    private $fbApp;
+    /**
+     * @var FacebookClient
+     */
+    private $fbSuccessClient;
+    /**
+     * @var FacebookClient
+     */
+    private $fbFailClient;
+
+    public function setUp()
+    {
+        $this->fbApp = new FacebookApp('app_id', 'app_secret');
+        $this->fbSuccessClient = new FacebookClient(new VideoUploadClientHandlerSuccess());
+        $this->fbFailClient = new FacebookClient(new VideoUploadClientHandlerFail());
+    }
+
+    public function testSuccess()
+    {
+        $fileUploader = new FacebookResumableUploader($this->fbApp, $this->fbSuccessClient, 'access_token');
+        $endpoint = '/113582528973300/videos';
+        $filePath = __DIR__ . '/../foo.mp4';
```

Comment:
I still don't think we need to include an actual video file in the SDK. We can mock this file. (Also another reason we need to pass in the `FacebookVideo` instance instead of the path to the file.)